packages/theme/src/cli/utilities/theme-fs.ts

```diff
@@ -190,13 +189,15 @@ export function mountThemeFileSystem(root: string, options?: ThemeFileSystemOpti
   const handleFileDelete = (themeId: string, adminSession: AdminSession, fileKey: string) => {
     if (isFileIgnored(fileKey)) return
 
-    unsyncedFileKeys.delete(fileKey)
+    // Optimistically delete the file from the local file system.
     files.delete(fileKey)
+    unsyncedFileKeys.add(fileKey)
     emitEvent('unlink', {fileKey})
 
     deleteThemeAsset(Number(themeId), fileKey, adminSession)
       .then(async (success) => {
         if (!success) throw new Error('Unknown issue.')
+        unsyncedFileKeys.delete(fileKey)
```

Comment:
With this change, there's now a new possible state: a file is not present in `themeFS.files` but it's still present in `themeFS.unsyncedFileKeys`. This signals that we had a file locally before but we have removed it optimistically while requesting the same thing to the remote SFR instance.